waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -47,10 +47,10 @@ export async function removeQueries() {
 
 export async function invalidateAndRemoveQueries() {
   const queryClient = await queryClientInitialized
-  // If we don't invalidate the queries before removing them, Wasp will stay on
+  // If we don't reset the queries before removing them, Wasp will stay on
   // the same page. The user would have to manually refresh the page to "finish"
   // logging out.
-  queryClient.invalidateQueries()
+  queryClient.resetQueries()
```

Comment:
This was a tricky one to get working, still not entirely sure about the difference between `invalidateQueries` and `resetQueries`.

From what I understood from the documentation, we should have always been using `resetQueries`, but `invalidateQueries` was working by accident.

It seems like `invalidateQueries` has been updating queries to their pre-loaded state until version 4 ([but wasn't supposed to](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#query-keys-and-mutation-keys-need-to-be-an-array)).

Anyway, I'll create a different issue for getting to the bottom of what's happening here, but this change matches our current behavior.

Reply:
Interesting, I remember we discussed this but couldn't quiet remember all the details. What you wrote sounds reasonable!